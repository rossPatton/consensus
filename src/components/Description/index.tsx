import React, {PureComponent} from 'react';

class Description extends PureComponent<
  {description: string},
  {showAll: boolean}
> {
  state = {
    showAll: false,
  };

  toggleDescription = () =>
    this.setState({
      showAll: !this.state.showAll,
    });

  render() {
    const {description} = this.props;
    const {showAll} = this.state;

    return (
      <>
        {!showAll && (
          <p className="mb-1">
            {description.split('\n')[0]}
          </p>
        )}
        {showAll
          && description.split('\n').map((p, i) => (
            <p
              key={i}
              className="mb-1">
              {p}
            </p>
          ))}
        <button
          className="mb-2 border-0 bg-0"
          onClick={this.toggleDescription}>
          {showAll ? 'Show less' : 'Show more'}
        </button>
      </>
    );
  }
}

export default Description;
