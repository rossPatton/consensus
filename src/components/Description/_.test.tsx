import React from 'react';
import render from 'react-test-renderer';

import Description from '.';

it('renders without crashing', () => {
  render.create(<Description />);
});

it('renders null when no description prop passed in', () => {
  const component = render.create(<Description />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders null when bad description passed in', () => {
  const component = render.create(<Description description="" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders as expected', () => {
  const component = render.create((
    <Description
      description="testing rendering description. test. testing."
    />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
