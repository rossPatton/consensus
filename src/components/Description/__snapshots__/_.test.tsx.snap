// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div
  className="break-words"
>
  <p
    className="mb-1"
  >
    testing rendering description. test. testing.
  </p>
</div>
`;

exports[`renders null when bad description passed in 1`] = `null`;

exports[`renders null when no description prop passed in 1`] = `null`;
